### PR TITLE
Pintpointer

### LINKED ISSUE #14617: Why would I take a pintpointer?
Icons for a new drink, the pintpointer.
pintpointer
Suggested in the thread, has different icons indicating proximity to the bar.
download link for icons: http://s000.tinyupload.com/index.php?file_id=09947391294335289252

@Shadowmech88 volunteered to code it, how nice of him.

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds a new drink recipe, the pintpointer. It can be made by mixing atomic bomb and Syndicate bomb.
image
Sprite courtesy of @Blithering.

If pintpointer is made in or poured into a glass held by someone, the display on the pintpointer will indicate its proximity to that person, ranging from full at <=8 tiles to empty at 41 tiles and beyond. If a pintpointer is made in or poured into a glass not held by someone, the display will simply reflect the amount of liquid in it.
Closes #14617 

:cl:
 * rscadd: Added a new drink recipe, the pintpointer. It can be made by mixing atomic bomb and Syndicate bomb. If pintpointer is made in or poured into a glass held by someone, the display on the pintpointer will indicate its proximity to that person.

### CODE REVIEW

**Comment by Exxion** [comment]
>atomic bomb and syndicate bomb

New record for most complex drink recipe, with, if my math/memory is right, seven mixing steps and eight separate ingredients, one of which is used twice
Remind me to not play for a few days after this gets merged

**Comment by Kurfursten** [comment]
Not only that, but continually alternating between 1:1, 2:1 and 1:1:1 mix ratios makes it a headache to find a formula that mixes evenly without leftovers

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm [hunk 1]
```diff
@@ -547,6 +547,14 @@
 					item_state = "ginvodkaglass"
 					name = "glass of Grey vodka"
 					desc = "A questionable concoction of objects found within maintenance. Tastes just like you'd expect."
+				if(PINTPOINTER)
+					var/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/pintpointer/P = new (get_turf(src))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm:551]: The problem with this is that if you mix it while it's in your hand you'll drop it on the floor which is not great.

Also, `get_turf()` is probably a bad choice.

**Comment by Hugh-G-Rection** [comment]
Is there a way to pour something into a glass held by someone else? Or am I going to have to trick security members into pouring it for themselves to make my redshirt proximity alarm?

**Comment by PJB3005** [comment]
Are grenades that do nothing re-usable? If so I could come up with
something.

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm (hunk 1 above)
Comment (Shadowmech88) [reply code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm:551]: I agree that it's not great, but I think it'd be worse to put all the Pintpointer code into `/drinkingglass` and just have it only function when the glass is full of pintpointer.

**Comment by Shadowmech88** [comment]
Is the recipe *too* complex?

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm (hunk 1 above)
Comment (PJB3005) [reply code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm:551]: Could at the very least do some checks to see if it's in the hand of a mob and swap it out of it is. Won't be fool proof but it's already an improvement.

Bonus points if you make it a generic proc so more code can do this.

**Comment by PJB3005** [comment]
I'd say that since the item is pretty powerful when used correctly, complexity is a good idea.

Keep in mind you only need the majority of a glass to be filled with it, so even making small amounts should be splittable among multiple glasses.

File: code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm (hunk 1 above)
Comment (Exxion) [reply code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm:551]: I did that but never pushed it
Just... leave it like this for now and I'll actually finish that system after my current PR is done

**Comment by Exxion** [comment]
Oh yeah, it's fairly useful, so it makes sense that it's very complex
It's just... very complex

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -5362,6 +5362,13 @@
 					H.adjustToxLoss(3)
 				H.adjustToxLoss(0.3)
 
+/datum/reagent/ethanol/deadrumm/pintpointer
+	name = "Pintpointer"
+	id = PINTPOINTER
+	description = "An attempt to create a navigation system which even a drunk spaceman can use."
```
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm:5368]: This description that is only ever visible when analyzed at a chemmaster is better than the description of the item itself

**Pull request merged**